addon/globalPlugins/charinfo/__init__.py

```diff
@@ -455,7 +455,8 @@ def displayCurrentCharInfoMessage(self, info):
 		### Code inspired from NVDA script_review_currentCharacter in file globalCommands.py
 		info.expand(textInfos.UNIT_CHARACTER)
 		if info.text == '':
-			speech.speakTextInfo(info,unit=textInfos.UNIT_CHARACTER,reason=controlTypes.REASON_CARET)
+			reasonCaret = getattr(controlTypes, "REASON_CARET", None) or controlTypes.OutputReason.CARET
```

Comment:
I prefer something more explicit and with the newest syntax first if possible (even if existing code has not always followed this requirement).
The following should work. Please double-check and confirm. Thanks.

```suggestion
			try:
				# Syntax for NVDA 2021.1 and above
				reasonCaret = controlTypes.OutputReason.CARET
			except AttributeError:
				# Syntax for NVDA 2020.4 and below
				reasonCaret = controlTypes.REASON_CARET)
```